tests/conftest.py

```diff
@@ -396,3 +401,111 @@ def log_cli_level(pytestconfig):
 
     """
     return pytestconfig.getini('log_cli_level') or 'DEBUG'
+
+
+@pytest.fixture(scope="session")
+def run_io_in_background(request):
+    """
+    Run IO during the test execution
+    """
+    if config.RUN['cli_params'].get('io_in_bg') and config.RUN['google_api_secret']:
+        log.info(f"Tests will be running while IO is in the background")
+
+        request.results = list()
+        request.temp_file = tempfile.NamedTemporaryFile(
+            mode='w+', prefix='test_status', delete=False
+        )
+
+        def get_test_status():
+            with open(request.temp_file.name, 'r') as t_file:
+                return t_file.readline()
+
+        def set_test_status(status):
+            with open(request.temp_file.name, 'w') as t_file:
+                t_file.writelines(status)
+
+        set_test_status('running')
+
+        def finalizer():
+            """
+            Delete the resources created during setup, used for
+            running IO in the test background
+            """
+            set_test_status('finished')
+            try:
+                for status in TimeoutSampler(90, 3, get_test_status):
+                    if status == 'terminated':
+                        break
+            except TimeoutExpiredError:
+                log.warning(
+                    "Background IO was still in progress before IO "
+                    "thread termination"
```

Comment:
I don't think this warning is technically correct.
it be reported when the thread self reported itself as running after request to shutdown during given timeout value. And you don't know if the thread will terminate at this point, so noting "before IO thread termination" is misleading (as after this, you are depending solely on thread self termination - which is btw correct procedure, just reported in a misleading way).

Reply:
the intention is to wait for the last iteration of io to finish before thread.join() which kills the IO thread and cause an ugly traceback to be printed